testhelper/docker/resource/mysql/mysql.go

```diff
@@ -0,0 +1,83 @@
+package mysql
+
+import (
+	"database/sql"
+	_ "encoding/json"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql" // mysql driver
+	"github.com/ory/dockertest/v3"
+	dc "github.com/ory/dockertest/v3/docker"
+
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
+)
+
+const (
+	mysqlDefaultDB       = "sources"
+	mysqlDefaultUser     = "root"
+	mysqlDefaultPassword = "password"
+)
+
+type Resource struct {
+	DB       *sql.DB
+	DBDsn    string
+	Database string
+	Password string
+	User     string
+	Host     string
+	Port     string
+}
+
+func Setup(pool *dockertest.Pool, d resource.Cleaner, opts ...func(*Config)) (*Resource, error) {
+	c := &Config{
+		Tag: "8.2",
+	}
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	// pulls an image, creates a container based on it and runs it
+	mysqlContainer, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "mysql",
+		Tag:        c.Tag,
+		Env: []string{
+			"MYSQL_ROOT_PASSWORD=" + mysqlDefaultPassword,
+			"MYSQL_DATABASE=" + mysqlDefaultDB,
+		},
+	}, func(hc *dc.HostConfig) {
+		hc.ShmSize = c.ShmSize
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	d.Cleanup(func() {
+		if err := pool.Purge(mysqlContainer); err != nil {
+			d.Log("Could not purge resource:", err)
+		}
+	})
+
+	dbDSN := fmt.Sprintf("%s:%s@tcp(127.0.0.1:%s)/%s?tls=false",
+		mysqlDefaultUser, mysqlDefaultPassword, mysqlContainer.GetPort("3306/tcp"), mysqlDefaultDB,
+	)
+	var db *sql.DB
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	err = pool.Retry(func() (err error) {
+		if db, err = sql.Open("mysql", dbDSN); err != nil {
+			return err
+		}
+		return db.Ping()
+	})
+	if err != nil {
+		return nil, err
```

Comment:
Would adding more context would be helpful?
```suggestion
		return nil, fmt.Errorf("could not connect to mysql container: %s", err)
```